web/src/test/auto/headless/engine/keyboard/keyboard-loading.tests.ts

```diff
@@ -0,0 +1,132 @@
+      const harness = new KeyboardInterface({}, MinimalKeymanGlobal);
+      const keyboardLoader = new NodeKeyboardLoader(harness);
+      const keyboard = await keyboardLoader.loadKeyboardFromPath(laoPath);
+      // --  END:  Standard Recorder-based unit test loading boilerplate --
+
+      // Asserts that the harness's loading field is cleared once the load is complete.
+      assert.isNotOk(harness.loadedKeyboard);
+
+      // Asserts that the `activeKeyboard` field was not set by the operation.
+      assert.isNotOk(harness.activeKeyboard);
+
+      // This part provides assurance that the keyboard properly loaded.
+      assert.equal(keyboard.id, "Keyboard_lao_2008_basic");
+    });
+
+    it('throws error when keyboard does not exist', async function () {
+      const harness = new KeyboardInterface({}, MinimalKeymanGlobal);
+      const keyboardLoader = new NodeKeyboardLoader(harness);
+
+      await assertThrowsAsync(() => keyboardLoader.loadKeyboardFromPath(nonExisting),
+        KeyboardDownloadError, `Unable to read keyboard file at ${nonExisting}`);
+    });
+
+    it('throws error when keyboard is invalid', async function () {
+      const harness = new KeyboardInterface({}, MinimalKeymanGlobal);
+      const keyboardLoader = new NodeKeyboardLoader(harness);
+
+      await assertThrowsAsync(() => keyboardLoader.loadKeyboardFromPath(nonKeyboardPath),
+        KeyboardMissingError, `Cannot find the keyboard at ${nonKeyboardPath}.`);
+    });
+
+    it('successfully loads (has variable stores)', async () => {
+      // -- START: Standard Recorder-based unit test loading boilerplate --
+      const harness = new KeyboardInterface({}, MinimalKeymanGlobal);
+      const keyboardLoader = new NodeKeyboardLoader(harness);
+      const keyboard = await keyboardLoader.loadKeyboardFromPath(ipaPath);
+      // --  END:  Standard Recorder-based unit test loading boilerplate --
+
+      // This part provides extra assurance that the keyboard properly loaded.
+      assert.equal(keyboard.id, "Keyboard_sil_ipa");
+    });
+
+    it('cannot evaluate rules', async function() {
+      // -- START: Standard Recorder-based unit test loading boilerplate --
+      const harness = new KeyboardHarness({}, MinimalKeymanGlobal);
+      const keyboardLoader = new NodeKeyboardLoader(harness);
+      const keyboard = await keyboardLoader.loadKeyboardFromPath(laoPath);
+      // --  END:  Standard Recorder-based unit test loading boilerplate --
+
+      // Runs a blank KeyEvent through the keyboard's rule processing...
+      // but via separate harness configured with a different captured global.
+      const ruleHarness = new KeyboardInterface({}, MinimalKeymanGlobal);
+      ruleHarness.activeKeyboard = keyboard;
+      assertThrows(() => ruleHarness.processKeystroke(new Mock(), keyboard.constructNullKeyEvent(device)), 'k.KKM is not a function');
+    });
+
+    it('accurately determines supported gesture types', async () => {
```

Comment:
I am wondering why this test is needed in the keyboard loading tests -- can you add comments to the code?